### DISCUSSION TITLE
Looking to create a PR for passive 'portscan' from censys data. Guidance on best way to do so.

### DISCUSSION THREAD

**Discussion by AkikoOrenji** [blacklanternsecurity/bbot#697] [Ideas]
Since bbot returns DNS_NAME data from the Censys certificate API  it might as well query the host API for input IP_ADDRESS and return OPEN_TCP_PORT providing a passive portscan capability. As best as i can tell this capability doesn't exist yet and overlaps with a standalone censys tool i've been writing.  Do you suggest i create a new module for this  e.g. censys-portscan.py  or should i look at modifying the existing censys module ? 
A new module would be easier for me however if you'd recommend it i'll roll up my sleeves and try to modify the existing.

**Comment by TheTechromancer** [comment]
Thanks for your interest in this! This is definitely something that could be added.

Actually, the censys module used to have this functionality, but it was removed -- partially due to some recurring problems with the poorly-maintained censys python library, and partially as part of the larger BBOT migration to asyncio.

You can see the old version of the module [here](https://github.com/blacklanternsecurity/bbot/blob/47a659aaaaedc5c6396b0e448677d4b304d042ce/bbot/modules/censys.py).

When I first wrote the module, the censys python library provided access to the other data types like open ports. But when the library was removed, we were left having to make direct web requests to their API, and I didn't have time to add in the other endpoints.

I think it would make sense to add this functionality back into the existing module. You should be able to use the module's old version as a starting point. It should only require a few modifications.

EDIT: I would also accept this is a separate module (`censys_portscan`). I don't see any problem with doing it that way, except maybe that the censys API Key would need to be added twice in the config. But the functionality between these two is different enough it seems like a separation would be sensible.
